docs/website/docs/plus/getting-started/advanced_tutorial.md

```diff
@@ -0,0 +1,305 @@
+
+### Accessing project settings
+
+A few examples of what you have access too on the project object:
+
+```py
+# show the currently active profile
+print(current.project().current_profile)
+# show the main project dir
+print(current.project().project_dir)
+# show the project config
+print(current.project().config)
+# list explicitely defined datasets (also works with destinations, sources, pipelines etc.)
+print(current.project().datasets)
+```
+
+### Accessing entities
+
+Accessing entities works in the same way as when accessing entities in the `dlt.yml`. If possible and allowed, implicit entities are created and returned back to you, if you, an error will be raised.
+
+```py
+# get a pipeline instance
+pipeline = current.entities().create_pipeline("my_pipeline")
+# get a destination instance
+destination = current.entities().create_destination("duckdb")
+```
+
+### Running pipelines with the predefined source with the runner
+
+`dlt+` provides a pipeline runner that is used when you run pipelines from the cli as described in the basic tutorial. The project context also provides a runner instance that you can use to run your pipelines in code. 
+
+```py
+# get the runner
+runner = current.runner()
+# run the pipeline "my_pipeline" from the currently active project
+runner.run_pipeline("my_pipeline")
+```
+
+### Accessing the catalog
+
+The catalog provides access to all explicitely defined datasets
+
+```py
+# get a dataset instance pointing to the default destination (first in dataset destinations list) and access data inside of it
+# for this to work this dataset must already exist physically
+dataset = current.catalog().dataset("my_pipeline_dataset")
+# get the row counts of all tables in the dataset as a dataframe
+print(dataset.row_counts().df())
+```
+
+:::tip
+Read the docs on how to access data in python in dlt datasets [here](../../../general-usage/dataset-access/dataset) to learn more about the available data access methods. You can browse and filter tables and get the data in various formats.
+:::
+
+### Writing data back to the catalog
+
+The datasets on the dlt+ catalogs may also be written to. The datasets provide `write` method that can be used to write data back to the catalog. You will be able to control which datasets may be written to with contracts in the future. Under the hood, `dlt+` will use an ad-hoc dlt pipeline to run this operation.
+
+:::warning
+Writing data back to the catalog is an experimental feature at this time and should be used with caution until it is fully stable.
+:::
+
+```py
+# get a dataset from the catalog (this dataset must already exist physically and be explicitely defined in the dlt.yml)
+dataset = current.catalog().dataset("my_pipeline_dataset")
+# write a dataframe to the dataset into the table "my_table"
+dataset.write(pd.DataFrame({"name": ["John", "Jane", "Jim"], "age": [30, 25, 35]}), table_name="my_table")
```

Comment:
We need to tell the reader to install and import Pandas as `pd` here, otherwise this sample won't work

Reply:
After installing Pandas and importing it as `pd` I get this error when running this line:

```
Traceback (most recent call last):
  File "./hackathon-0228/do_something.py", line 31, in <module>
    dataset.write(pd.DataFrame({"name": ["John", "Jane", "Jim"], "age": [30, 25, 35]}), table_name="my_table")
TypeError: 'ReadableDBAPIRelation' object is not callable
```

Reply:
same for me. I also agree that we should just add an `import pandas as pd` to the code, which the user can then move to the top of their file after having copied it over